### PR TITLE
feat: Add Support for creating user entries with alias flag

### LINKED ISSUE #5164: EKS update-config should support naming the user with the passed in alias
**Is your feature request related to a problem? Please describe.**
I have multiple aws role to k8s role mappings per cluster. The eks update-config command supports setting the `AWS_PROFILE` variable under the user config if you pass in `--profile`. However, it defaults the username to the cluster arn, so if I do this with multiple roles it gets overwritten  

**Describe the solution you'd like**
There is already precedent for overriding the context name by passing an `--alias`. If I run the following:
`aws eks update-kubeconfig --name <clustername> --alias prod-admin --profile prod-admin-role` 

AND

`aws eks update-kubeconfig --name <clustername> --alias prod-support --profile prod-support-role` 

I would like for the following config to be created:

```
- context:
    cluster: arn:aws:eks:us-west-2:redacted:cluster/prod-cluster
    user: prod-admin
  name: prod-admin
- context:
    cluster: arn:aws:eks:us-west-2:redacted:cluster/prod-cluster
    user: prod-support
  name: prod-support
kind: Config
preferences: {}
users:
- name: prod-admin
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - us-west-2
      - eks
      - get-token
      - --cluster-name
      - prod-cluster
      command: aws
      env:
      - name: AWS_PROFILE
        value: prod-admin-role
- name: prod-support
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - us-west-2
      - eks
      - get-token
      - --cluster-name
      - prod-cluster
      command: aws
      env:
      - name: AWS_PROFILE
        value: prod-support-role
```

This will allow me to simply run a command to loop through all my roles, list eks-clusters, and define contexts for each cluster and role.

**Describe alternatives you've considered**
I could create a separate utility to do this, but honestly even forking this repo and making the small change would be easier. 

**Additional context**

I have this working locally, and it is a 2 line change to match existing functionality with context naming.

### PR DESCRIPTION
**Pull request by elijah-roberts**
*Issue #, if available: Fixes #5164

*Description of changes: Added support for creating user name in kube-config with alias (simliar to the way context works with alias)
Added appropriate testing for user name creation with and without alias flag

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by elijah-roberts** [comment]
@kyleknap @elysahall  Anything else I can do for this PR?

**Comment by nckturner** [comment]
@elijah-roberts in your example from the issue, the context name is set to the same value as the user name.  Is that missing from this PR?

**Comment by nckturner** [comment]
Nvm, I think I see how it works now, alias was already provided as a flag to set context name.  While using alias for both username and context name would work, it seems like we could add a `--user` flag and not worry about breaking any users relying on the current behavior.

**Comment by kdaily** [comment]
Hi @elijah-roberts, can you address @nckturner's suggestions?

**Comment by elijah-roberts** [comment]
> Nvm, I think I see how it works now, alias was already provided as a flag to set context name. While using alias for both username and context name would work, it seems like we could add a `--user` flag and not worry about breaking any users relying on the current behavior.

That would work for me 👍  I can plan on implementing a flag and testing that out.

**Comment by elijah-roberts** [comment]
@nckturner @kdaily  I appreciate your reviewing my PR. I updated with the changes you suggested by breaking out this logic into a new flag `user-alias`. I have updated tests, and confirmed that it is working as expected for myself.

**Comment by elijah-roberts** [comment]
@nckturner @kdaily  Looks like the codecov/project failed despite the fact that I added testes for the code that I added. Is there something else that I need to do?

**Comment by kdaily** [comment]
@nckturner - let me know here if this is good to go!

@elijah-roberts - I'll look into the coverage failure.

Thanks both!

**Comment by elijah-roberts** [comment]
Bump... @kdaily  Any update on this?

**Comment by nckturner** [comment]
@kdaily this LGTM

**Review by nckturner — APPROVED** [review]

**Comment by potto007** [comment]
My team could really use this feature. @kdaily is this going to be merged?

Thanks!

**Comment by joaoubaldo** [comment]
+1 waiting for this to be merged. Any updates @kdaily?

**Comment by kdaily** [comment]
@joaoubaldo, @potto007, @elijah-roberts - this feature still needs review from the CLI team, and it is on the agenda for review. Marking it as such now. It's on the radar, but I don't know when it will be able to pull in.

**Comment by joshdk** [comment]
+1 - This feature would be a welcome addition

**Comment by elijah-roberts** [comment]
@joaoubaldo, @potto007, @kdaily bumping again, this is a simple non-breaking change that is useful for supporting the creation of multiple kube-configs for different roles. For enterprise EKS users typically there will be a daily role that has limited permissions, and mappings. There will also be a break-glass type admin role. Current state to support both of these you have to manually manage your kubeconfig file, or constantly overwrite the associated user each time you update the config with a different role. 

Any chance we can get a review on this ? I am happy to make any necessary changes.

**Comment by elijah-roberts** [comment]
Pulled in latest changes from develop, and the code coverage report looks good now

**Review by yuxiang-zhang — APPROVED** [review]

**Comment by yuxiang-zhang** [comment]
We are trying to prioritize this feature request and get it merged soon. Note that this implements the same feature as #5413

**Comment by joshuamkite** [comment]
> We are trying to prioritize this feature request and get it merged soon. Note that this implements the same feature as #5413

Thanks but that issue has also been open for the best part of 3 years. What's the holdup?

**Review by stealthycoin — APPROVED** [review]

**Pull request merged**
